lib/mongo_client.js

```diff
@@ -188,7 +188,7 @@ Object.defineProperty(MongoClient.prototype, 'writeConcern', {
 Object.defineProperty(MongoClient.prototype, 'readPreference', {
   enumerable: true,
   get: function() {
-    return ReadPreference.primary;
+    return ReadPreference.fromOptions(this.s.options) || ReadPreference.primary;
```

Comment:
This is a property that will likely be checked very often (most operations), I think a better approach would be to resolve the `ReadPreference.fromOptions` in the constructor, save it as private/symbol property of the `MongoClient` class itself, and return the cached value here instead of computing it each time. 

Also, to answer your question on the main PR: yes, there should be tests for this change

Reply:
Hey Matt! Great idea caching this! Added cache + test.